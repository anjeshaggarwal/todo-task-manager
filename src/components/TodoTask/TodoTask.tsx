import React from 'react'
import './TodoTask.css'

interface IProps {
    text: string
    status: boolean
    onClick: (text: string) => void
}

const TodoTask = React.memo((props: IProps) => {
    return <li data-testid={"list-item"} className={props.status ? "strike" : ""}><span data-testid="list-span" onClick={(event: any) => props.onClick(event.target.innerHTML)}>{props.text}</span></li>
}, (prevProps: IProps, nextProps: IProps) => prevProps.status === nextProps.status)

export default TodoTask;