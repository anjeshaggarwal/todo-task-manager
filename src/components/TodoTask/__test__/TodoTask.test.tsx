import ReactDOM from 'react-dom'
import TodoTask from './../TodoTask'
import { render, cleanup, fireEvent } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer'

afterEach(cleanup)

it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<TodoTask text="Test" status={true} onClick={jest.fn} />, div)
})

it('renders correctly',()=>{
    const {getByTestId} = render(<TodoTask text="Test" status={true} onClick={jest.fn} />)
    expect(getByTestId('list-item')).toBeTruthy()
})

it('match snapshot with striked text',()=>{
    const tree = renderer.create(<TodoTask text="Test" status={true} onClick={jest.fn} />).toJSON()
    expect(tree).toMatchSnapshot()
})

it('match snapshot without striked text',()=>{
    const tree = renderer.create(<TodoTask text="Test" status={false} onClick={jest.fn} />).toJSON()
    expect(tree).toMatchSnapshot()
})

it('function call on click',()=>{
    const clickFunction = jest.fn()
    const {getByTestId} = render(<TodoTask text="Test" status={true} onClick={clickFunction} />)
    const span = getByTestId('list-span')
    expect(span).toBeTruthy()
    fireEvent.click(span)
    expect(clickFunction).toHaveBeenCalled()
})

it('function not called on click of li',()=>{
    const clickFunction = jest.fn()
    const {getByTestId} = render(<TodoTask text="Test" status={true} onClick={clickFunction} />)
    const li = getByTestId('list-item')
    expect(li).toBeTruthy()
    fireEvent.click(li)
    expect(clickFunction).not.toHaveBeenCalled()
})