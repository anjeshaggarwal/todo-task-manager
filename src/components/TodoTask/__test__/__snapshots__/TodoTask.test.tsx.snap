// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot with striked text 1`] = `
<li
  className="strike"
  data-testid="list-item"
>
  <span
    data-testid="list-span"
    onClick={[Function]}
  >
    Test
  </span>
</li>
`;

exports[`match snapshot without striked text 1`] = `
<li
  className=""
  data-testid="list-item"
>
  <span
    data-testid="list-span"
    onClick={[Function]}
  >
    Test
  </span>
</li>
`;
